platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -923,6 +932,97 @@ class ConfigurationCacheFingerprintWriter(
     override fun onScriptFileResolved(scriptFile: File) {
         fileObserved(scriptFile)
     }
+
+    override fun onGradlePropertiesByPrefix(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        prefix: String,
+        snapshot: Map<String, String>
+    ) {
+        if (isInputTrackingDisabled()) {
+            return
+        }
+        if (gradlePropertiesByPrefix.computeIfAbsent(propertyScope) { newConcurrentHashSet() }.add(prefix)) {
```

Comment:
💅  It's a somewhat heavy-handed if expression, should we extract it into a function?